clippy_lints/src/methods/mod.rs

```diff
@@ -1498,11 +1498,20 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Methods {
                 if cx.access_levels.is_exported(impl_item.hir_id) {
                 // check missing trait implementations
                     for &(method_name, n_args, fn_header, self_kind, out_type, trait_name) in &TRAIT_METHODS {
+                        let no_lifetime_params = || {
+                            impl_item.generics.params.iter().filter(|p| match p.kind {
+                               hir::GenericParamKind::Lifetime { .. } => true,
```

Comment:
Did not test it, but I think elided lifetimes would be catched here according to the [docs](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.LifetimeParamKind.html) as all kinds are covered

Reply:
Indeed, I didn't spot that. 

But somehow the elided kind doesn't show up in the tests. All the elided test cases are linted although we filter out methods with lifetime params of any kind.

The only test method with explicit lifetime params is now filtered out. But whether we want to do that for all methods with explicit lifetime params still needs to be discussed.